### ISSUE TITLE
[Ray Core] Ray nightly GPU docker image broken on NVIDIA V100 GPUs on AWS

### ISSUE THREAD

**Issue by jaanphare** [ray-project/ray#43565]
### What happened + What you expected to happen

1. This template works on NVIDIA A10 GPUs on AWS (g5.xlarge instances): https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml
2. Changing 1 line in this configuration (instance type: g5.xlarge to p3.2xlarge) leads to this command failing: `python -c "import torch; print(torch.cuda.is_available())"` (on g5.xlarge instances it runs fine). 

The base AMI is the NVIDIA AMI (ami-041855406987a648b). 

SSHing into the node directly and running `python -c "import torch; print(torch.cuda.is_available())"` works fine, which demonstrates the problem is isolated to the ray docker image. 

So far all of these fail:
```
rayproject/ray-ml:latest-gpu
rayproject/ray:nightly-py310-cu121
rayproject/ray-ml:latest-py39-cu118
```

Any ideas for debugging this? Thank you!!

### Versions / Dependencies

```
rayproject/ray-ml:latest-py39-cu118
rayproject/ray:nightly-py310-cu121
rayproject/ray-ml:latest-gpu
```

p3.2xlarge and g5.xlarge instances on AWS with https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml.

### Reproduction script

`ray up https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml`
`python -c "import torch; print(torch.cuda.is_available())"`

### Issue Severity

High: It blocks me from completing my task.

**Comment by jaanphare** [comment]
Here is the cluster configuration if needed: 

```
cluster_name: debug_p3_instances_with_ray

upscaling_speed: 1.0
max_workers: 10
idle_timeout_minutes: 5

docker:
    image: "rayproject/ray:2.9.1-py39-cu121"
    container_name: "ray_container"
    pull_before_run: True
    run_options: 
        - --ulimit nofile=65536:65536

initialization_commands: []

provider:
    type: aws
    region: us-east-1
    availability_zone: us-east-1a, us-east-1b, us-east-1c, us-east-1d, us-east-1e, us-east-1e

auth:
    ssh_user: ubuntu

available_node_types:
    ray.head.default:
        resources: {}
        node_config:
            InstanceType: p3.2xlarge
            ImageId: ami-041855406987a648b
            BlockDeviceMappings:
                - DeviceName: /dev/sda1
                  Ebs:
                      VolumeSize: 140
                      VolumeType: gp3
                      Throughput: 500
    ray.worker.default:
        min_workers: 4
        max_workers: 10
        resources: {}
        node_config:
            InstanceType: p3.2xlarge
            ImageId: ami-041855406987a648b
            BlockDeviceMappings:
                - DeviceName: /dev/sda1
                  Ebs:
                      VolumeSize: 140
                      VolumeType: gp3
                      Throughput: 500

head_node_type: ray.head.default
file_mounts: {}
cluster_synced_files: []
file_mounts_sync_continuously: False

setup_commands:
    - pip3 install --upgrade pip
    - pip3 uninstall -y torch torchvision torchaudio
    - pip3 install torch torchvision torchaudio
    - pip3 install numpy
    - pip3 install torcheval 
    - pip3 install jaxtyping 
    - pip3 install beartype 
    - pip3 install omegaconf
    - pip3 install wandb
    - pip3 install tokenizers
    - pip3 install scikit-learn
    - pip3 install scipy 
    - pip3 install transformers 
    - pip3 install boto3
    - python -c "import torch; print('Torch version - ', torch.__version__)"
    - python -c "import torch; assert torch.cuda.is_available(), 'CUDA not available'"
head_setup_commands: []
worker_setup_commands: []
head_start_ray_commands:
    - ray stop
    - ray start --head --port=6379 --object-manager-port=8076 --autoscaling-config=~/ray_bootstrap_config.yaml --dashboard-host=0.0.0.0
worker_start_ray_commands:
    - ray stop
    - ray start --address=$RAY_HEAD_IP:6379 --object-manager-port=8076
```

This fails at the assert statement for `torch.cuda.is_available()`.

**Comment by justinvyu** [comment]
@can-anyscale Are you aware of this issue related to our OSS GPU images?

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
